api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -1558,4 +1560,132 @@ public CohortDefinition getBreastfeedingPatientsStartingART() {
+            hivMetadata.getSingleStop().getConceptId(),
+            hivMetadata.getStartDrugs().getConceptId(),
+            hivMetadata.getContinueRegimen().getConceptId()));
+    return sql;
+  }
+
+  /**
+   * Number of active patients on ART and are marked in the last PU as I OR C on Ficha Clinica and
+   * are aligible
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition
+      getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinicaAndEligible() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName(
+        "Number of active patients on ART who are in PU as I or C on ficha clinica and are eligible");
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addSearch(
+        "TxCurrPu",
+        EptsReportUtils.map(
+            getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinica(),
+            "endDate=${endDate},startDate=${startDate},location=${location}"));
+    cd.addSearch(
+        "Active",
+        EptsReportUtils.map(
+            getAllPatientsWhoAreActiveAndStable(),
+            "endDate=${endDate},startDate=${startDate},location=${location}"));
+    cd.setCompositionString("TxCurrPu AND Active");
+
+    return cd;
+  }
+
+  /**
+   * Number of active patients on ART and are marked in the last PU as I OR C on Ficha Clinica and
+   * are NO Eligible
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition
+      getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinicaAndNotEligible() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName(
+        "Number of active patients on ART who are in PU as I or C on ficha clinica and are NOT eligible");
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addSearch(
+        "TxCurrPu",
+        EptsReportUtils.map(
+            getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinica(),
+            "endDate=${endDate},startDate=${startDate},location=${location}"));
+    cd.addSearch(
+        "NotActive",
+        EptsReportUtils.map(
+            getPatientsWhoAreActiveAndUnstable(),
+            "endDate=${endDate},startDate=${startDate},location=${location}"));
+    cd.setCompositionString("TxCurrPu AND NotActive");
+
+    return cd;
+  }
+
+  /**
+   * Number of active patients on ART and are marked in the last PU as I OR C on Ficha Clinica and
+   * are aligible and NOT on TB treatment
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition
+      getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinicaAndEligibleAndNotOnTbtreatment() {
```

Comment:
This method is unnecessary.  The PR for **Ticket no. 287 - D1**(#153 ) has already addressed the issue of filtering Patients Not on TB Treatment for Stable and Unstable patients.  
Therefore, the method `getNumberOfPatientsOnArtAndAreMArkedInLastPuAsIorConFichaClinicaAndEligible()` can sufficiently be used in place of this one.

Reply:
Done